server/graphql/core/src/loader/item_direction.rs

```diff
@@ -0,0 +1,62 @@
+use actix_web::web::Data;
+use repository::{
+    item_direction::{ItemDirection, ItemDirectionFilter, ItemDirectionRepository},
+    EqualFilter, RepositoryError,
+};
+
+use async_graphql::dataloader::*;
+use service::service_provider::ServiceProvider;
+use std::collections::HashMap;
+
+pub struct ItemDirectionsByItemIdLoader {
+    pub service_provider: Data<ServiceProvider>,
+}
+
+impl Loader<String> for ItemDirectionsByItemIdLoader {
+    type Value = Vec<ItemDirection>;
+    type Error = RepositoryError;
+
+    async fn load(&self, item_ids: &[String]) -> Result<HashMap<String, Self::Value>, Self::Error> {
+        let service_context = self.service_provider.basic_context()?;
+        let repo = ItemDirectionRepository::new(&service_context.connection);
+
+        let item_directions = repo.query_by_filter(
+            ItemDirectionFilter::new().item_id(EqualFilter::equal_any(item_ids.to_vec())),
+        )?;
+
+        let mut map: HashMap<String, Vec<ItemDirection>> = HashMap::new();
+        for direction in item_directions {
+            let list = map.entry(direction.item_row.id.clone()).or_default();
+            list.push(direction);
+        }
+
+        Ok(map)
+    }
+}
+
+pub struct ItemDirectionByItemDirectionIdLoader {
+    pub service_provider: Data<ServiceProvider>,
+}
+
+impl Loader<String> for ItemDirectionByItemDirectionIdLoader {
```

Comment:
This isn't actually used and is here for boiler plate cargo cult reasons

Reply:
Does this translate to "my next PR will use it"?

Is long term that a prescription would store item_direction_id and directions would be loaded via this?

Otherwise burn it lol

Reply:
No I see no use case for it, that is the only feasible idea long term that might and it ain't one we wanna do.

Reply:
https://github.com/msupply-foundation/open-msupply/pull/6408/commits/f03fa967ad7a81e9e48b21dd09516ee9551638ff